src/step-function.ts

```diff
@@ -47,6 +47,16 @@ export type AnyStepFunction =
   | ExpressStepFunction<any, any>
   | StepFunction<any, any>;
 
+export type StepFunctionClosure<
+  Payload extends Record<string, any> | undefined,
+  Out
+> = (arg: Payload) => Promise<Out> | Out;
+
+type ParallelFunction<T> = () => Promise<T> | T;
```

Comment:
Why not always a Promise

Reply:
It allows for functions that don't return promise.

```ts
() => 1 + 2 // this fails without | T because it doesn't return a promise.
() => func() // promise
async () => {
    const await func();
}
```